connect/runtime/src/main/java/org/apache/kafka/connect/storage/ConnectorOffsetBackingStore.java

```diff
@@ -279,8 +284,60 @@ public Future<Void> set(Map<ByteBuffer, ByteBuffer> values, Callback<Void> callb
             throw new IllegalStateException("At least one non-null offset store must be provided");
         }
 
+        boolean containsTombstones = values.containsValue(null);
+
+        // If there are tombstone offsets, then the failure to write to secondary store will
+        // not be ignored. Also, for tombstone records, we first write to secondary store and
+        // then to primary stores.
+        if (secondaryStore != null && containsTombstones) {
+            AtomicReference<Throwable> secondaryStoreTombstoneWriteError = new AtomicReference<>();
+            FutureCallback<Void> secondaryWriteFuture = new FutureCallback<>((secondaryWriteError, ignored) -> {
```

Comment:
Why do we need this underlying callback to handle `secondaryWriteError`? Can't we just use a plain `FutureCallback` instantiated via the default constructor ? The error will anyway be surfaced synchronously when we call `get` on our `FutureCallback` (and handled in the `ExecutionException` catch block)?

(Also mentioned in https://github.com/apache/kafka/pull/13801#discussion_r1255451696, but that comment was resolved and I don't seem to have permissions to un-resolve it).

Reply:
yeah, that makes sense. Sorry didn't realise I don't even need that. I have made the changes.